server/room-battle.ts

```diff
@@ -878,8 +878,16 @@ export class RoomBattle extends RoomGames.RoomGame {
 		const logfolder = logsubfolder.split('-', 2).join('-');
 		const tier = this.room.format.toLowerCase().replace(/[^a-z0-9]+/g, '');
 		const logpath = `logs/${logfolder}/${tier}/${logsubfolder}/`;
+
+		let logID = this.roomid;
+		if (logID.endsWith('pw')) {
```

Comment:
I think you might want to use the `getReplayData` function from https://github.com/smogon/pokemon-showdown/issues/6950#issuecomment-653822792? In general Zarel and I discussed wanting a single method for parsing the pw-ed ids

Reply:
Agree with pre on this one.